botocore/endpoint.py

```diff
@@ -38,6 +38,20 @@
 filter_ssl_warnings()
 
 
+def create_endpoint_url(endpoint_config, is_secure):
```

Comment:
I would be in favor of not adding any of this logic to ``endpoint`` module. Anything to simplify/remove logic from this module is ideal for me. Maybe make it a part of the helper default value determiner class I was talking about?